### DISCUSSION TITLE
Fief as user authenticator for streamlit app

### DISCUSSION THREAD

**Discussion by vpatel289** [fief-dev/fief#148] [General]
Is there any tutorial or example for how to use the Fief as user authenticator for Streamlit or is possible to use Fief with streamlit?

**Comment by fief-bailiff** [comment]
Hail, @vpatel289 👋 Welcome to Fief's kingdom!

Our team will get back to you very soon to help.

In the meantime, take a minute to star our repository ⭐️

![star-fief](https://user-images.githubusercontent.com/1144727/192523552-dc2c31fa-ae94-42fa-9ab4-7d0ba51bb7b0.png)

Farewell!

**Comment by frankie567** [comment]
Hi @vpatel289 👋 

Unfortunately, we do not have any guidelines right now for Streamlit. You should however be able to use our [base Python client](https://docs.fief.dev/integrate/python/). The key things to do would be:

1. Generate the authorization URL and make your Streamlit app redirect to it.
2. Handle the callback in your Streamlit app by retrieving the `code` in the URL query parameters.
3. Save the user data and tokens somewhere, I think Streamlit may have some session state.
